core/network/impl/synchronizer_impl.cpp

```diff
@@ -59,16 +61,69 @@
     app_state_manager->atShutdown([this] { node_is_shutting_down_ = true; });
   }
 
+  bool SynchronizerImpl::subscribeToBlock(
+      const primitives::BlockInfo &block_info, SyncResultHandler &&handler) {
+    // Check if block is already in tree
+    if (block_tree_->hasBlockHeader(block_info.hash)) {
+      handler(block_info);
+      return false;
+    }
+
+    auto last_finalized_block = block_tree_->getLastFinalized();
+    // Check if block from discarded side-chain
+    if (last_finalized_block.number <= block_info.number) {
+      handler(Error::DISCARDED_BLOCK);
+      return false;
+    }
+
+    // Check if block has arrived too early
+    auto best_block_res =
+        block_tree_->getBestContaining(last_finalized_block.hash, std::nullopt);
+    BOOST_ASSERT(best_block_res.has_value());
+    const auto &best_block = best_block_res.value();
+    if (best_block.number + 512 < block_info.number) {
```

Comment:
magic numbers..

Reply:
I'll make reasonable constants with explanation in comments. Thx.